tests/test.rs

```diff
@@ -18,7 +29,7 @@ fn create_socketpair() -> (Socket, Socket) {
     receiver.set_rcvtimeo(1000).unwrap();
 
     receiver.bind("tcp://*:*").unwrap();
```

Comment:
See my comment in #141; it seems the issue is not with `get_last_endpoint()`, which behaves identically (and sanely, IMHO) on both platforms, IIUC. The question is rather if a `connect()` to `INADDR_ANY` really is well-defined even across POSIX-like systems.

What about not using the wildcard address in the tests at all? I see
only downsides of binding any sockets in the test suite to the wildcard address:

- It allocates resources more broadly than needed on the machine running the tests.
- It has security implications (also binding to the loopback has security implications, see e.g. [CVE-2016-8606](http://seclists.org/oss-sec/2016/q4/100). Of course the attack surface is much smaller here ;-).
- It makes us deal with platform-specifica -- namely how to find out the peer's IP address for no good reason.

Even when the libzmq test suite does it with `#ifdef` and wildcard addresses, we don't have to emulate that ;-p.

Anyway, this is a rather minor point, the important thing is you found the underlying cause (some of?) our Windows test suite failures, @pfernie!.